src/LEGO.AsyncAPI.Readers/Interface/IAsyncApiExternalReferenceReader.cs

```diff
@@ -0,0 +1,14 @@
+namespace LEGO.AsyncAPI.Readers;
+
+/// <summary>
+/// Interface that provides method for reading external references.å
+/// </summary>
+public interface IAsyncApiExternalReferenceReader
+{
+    /// <summary>
+    /// Method that returns the AsyncAPI content that the external reference from the $ref points to.
+    /// </summary>
+    /// <param name="reference">The content address of the $ref.</param>
+    /// <returns>The content of the reference as a string.</returns>
+    public string GetExternalResource(string reference);
```

Comment:
Id love for this to be named something like `Load` and similarly have an addition "LoadAsync" method.
Then we can simply create an async LoadReferenceAsync, for when using http or similar ingestion of external resources.

Reply:
I'm not sure how async would work as ResolveReferences could is all implemented synchronously. Would this require creation of a asynchronous reference resolution class?

Reply:
@VisualBean Yeah, I'm definitely onboard with it is a good feature, just want to clarify the complexity of the implementation and scope. I've had a stab at it and its substantially increasing the amount of code in this PR.

A new `AsyncApiVisitorBaseAsync` needs to be created with all the 35 `Visit` functions in their `async` variant as `VisitAsync` and then implementations in an async external reference reader and dispatch from the enclosing `ReadAsync` methods in the readers. This would be fine but I realised after getting that far that there is also no `async` variant of the fragment readers so once you've resolved one reference further recursive references will all be synchronous which doesn't make sense if all the references are stored remotely. This would really start to encroach on updating the core reader functionality and I would want to test and QA that stuff properly. I think there's even a design discussion to be had for users who have a mix of synchronous and asynchronous resources. Also, I haven't even reached the end of actually solving the implementation and its already ballooning. There could very well be more places I need to make changes.

All this to say I think it's potentially going to hold up getting basic external reference reading out and usable and providing value now if we offer the async path. I want to do the async properly but also don't want to end up blocking this work with a particularly large PR. So, perhaps it would be better to leave the `LoadAsync` for a follow PR and record it as a coming feature? What do you think?

Reply:
Hmm yeah I get that.

Maybe I can add it later with the jsonschema change, so we don't need to change the entirety of the flow to be Async, but rather only the document loading. 

Let's keep as is, and we can build it out in a later version

Reply:
Great, I'm happy to be involved with making it happen. I already sketched out some of it!